esp32-wroom-rp/src/spi.rs

```diff
@@ -317,21 +318,24 @@ where
             return Err(ProtocolError::InvalidNumberOfParameters.into());
         }
 
-        let num_params_to_read = self.get_byte().ok().unwrap() as usize;
+        let number_of_params_to_read = self.get_byte().ok().unwrap() as usize;
 
-        // TODO: use a constant instead of inline params max == 8
-        if num_params_to_read > 8 {
+        if number_of_params_to_read > MAX_NINA_PARAMS {
             return Err(ProtocolError::TooManyParameters.into());
         }
 
-        let mut params: [u8; ARRAY_LENGTH_PLACEHOLDER] = [0; ARRAY_LENGTH_PLACEHOLDER];
-        for (index, _param) in params.into_iter().enumerate() {
-            params[index] = self.get_byte().ok().unwrap()
+        let mut response_param_buffer: [u8; MAX_NINA_RESPONSE_LENGTH] =
+            [0; MAX_NINA_RESPONSE_LENGTH];
+        if number_of_params_to_read > 0 {
+            for (index, _) in response_param_buffer.into_iter().enumerate() {
+                response_param_buffer[index] = self.get_byte().ok().unwrap()
+            }
         }
+
         let control_byte: u8 = ControlByte::End as u8;
         self.read_and_check_byte(&control_byte).ok();
```

Comment:
This is a part of the communication where we should decide how we want to handle things. The way this is and has been written we check for a `ControlByte::End` (`0xEE`) and then do nothing regardless of what `read_and_check_byte` returns. 

Does it make sense to return an error if we do not receive the `ControlByte::End`. 
I would think so because it either means we did not read the entire response or we read extra bytes.

Reply:
This can be addressed in the next PR.